File: leetcode/605. Can Place Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        if(n == 0){
            return true;
        }
        
        for(size_t i = 0; i < flowerbed.size(); ++i){
            if(flowerbed[i] == 1){
                continue;
            }
            
            if((i == 0 || flowerbed[i - 1] == 0) && (i == flowerbed.size() - 1 || flowerbed[i + 1] == 0)){
                flowerbed[i] = 1;
                n -= 1;
                if(n == 0){
                    return true;
                }
                    
            }
        }
        return false;
    }
};

File: leetcode/605. Can Place Flowers.rs
impl Solution {
    pub fn can_place_flowers(mut flowerbed: Vec<i32>, n: i32) -> bool {
        if n == 0 {
            return true;
        }
        let mut cnt = 0;
        for i in (0..flowerbed.len()) {
            if flowerbed[i] == 0 {
                if (i == 0 || flowerbed[i-1] == 0) && (i == flowerbed.len()-1 || flowerbed[i+1] == 0) {
                    cnt += 1;
                    if cnt >= n {
                        return true;
                    }
                    flowerbed[i] = 1;
                }
            }
        }
        
        false
    }
}
